### PR TITLE
Add default voting methods

### PR DESCRIPTION
**Pull request by cap10morgan**
I did most of this as a Friday project so we wouldn't have to hard-code a state-by-state data structure of voting ID instructions in election-notification-works.

This dovetails nicely with #9 (it needs some of the code in here). So once this is merged to master, I will merge master into #9 and resolve any conflicts.

### CODE REVIEW

File: src/election_authority_works/entities/authority.clj [hunk 1]
```diff
@@ -4,6 +4,21 @@
...
+   :instruction/text text})
+
+(defn voting-method->tx-data
+  [{:keys [type name instructions] :as voting-method}]
+  (when voting-method
+    (db/dissoc-optional-nil-values
+     {:voting-method/type type
+      :voting-method/name name
+      :voting-method/instruction (when (seq instructions)
+                                   (map instruction->tx-data instructions))}
```
Comment (tie-rack) [inline src/election_authority_works/entities/authority.clj]: Seems like the `when` is unnecessary. `(map f nil)` is `nil`. In fact, one of the first things the 2-arity `map` does is a `when-let` on `seq` of the collection argument.

**Comment by tie-rack** [comment]
The `:instruction/text` and `:voting-method/name` attributes and how they're used seem to press us into a situation where translations are going to have to be bolted on instead of integrated seamlessly. The translation chicken is going to look for a place to land, and I'm afraid we're not giving it much of a ledge.

For that reason, I'd like to propose either leaving those attributes out entirely (letting a future effort deal with user-facing language—which, maybe doesn't belong in this component?), or making the English-ness of the current values explicit (by making those attributes refs to entities that have both `text` and `language` attributes).

**Comment by cap10morgan** [comment]
@tie-rack I don't see how this precludes us from nor encourages us in any particular future direction w/r/t i18n. It seems like we could just add new attributes for the i18n keys and use those + the user's locale to look up the translation. Unless I'm missing something.

My fear with what you propose is that it will lead us to thinking i18n is implemented and we can start putting other languages in and using it. But it's not clear that this solution will cover all our use cases for i18n, and I don't want to design that in here.

And we need `:instruction/text` soon. We could potentially leave out `:voting-method/name`, I suppose. But I'd rather just migrate it to the i18n system once that's implemented.

**Comment by tie-rack** [comment]
Are you suggesting adding `:instruction/text-es`, `:instruction/text-fr`, etc. as alternates to `:instruction/text`? In that case, English has been given (I'd argue) unnecessarily special treatment.

The only thing I'm suggesting is a manner of _storing_ the language explicitly with the text, so that later work of actually dealing with internationalization doesn't require wholesale changes to how text is stored, extensive changes to code dealing with reading and writing that text.

What language a piece of text belongs to is unquestionably important to any internationalization system we eventually invent. We're going to have to store it with the text. Adding suffixed attributes for each instance of a stored piece of text (`text-es`, `text-jp`, `text-ch`... _and_ `name-es`, `name-jp`, `text-ch`...) falls on the wrong side of the expression problem.

**Comment by cap10morgan** [comment]
No, I'm not suggesting that. I'm giving an example of adding a new attribute like `:instruction/text-i18n` and putting the i18n lookup key in there as it's value. That + the user's locale can then be used to retrieve the localized text from a hypothetical i18n system.

But that was just an example to demonstrate that I think we still have lots of i18n options and that the approach currently in this code is intentionally not attempting to implement (nor preclude) any of them.

Ultimately I don't want to assume anything about how i18n will work in here.

**Comment by tie-rack** [comment]
I can see that, but putting user-facing English text in a field called `text` and relegating other languages to some other system is anti-internationalization. My first suggestion of removing user-facing text here stands.

**Comment by cap10morgan** [comment]
@tie-rack I'm agree and I'm not suggesting that either. What I'm suggesting is this:

* As of right now, our system only supports English. Hopefully that's not a surprise to anyone.
* We need to build an i18n system soon, and once we do, we move *every* language, including English into that system. There will be no privileged languages.
* We make whatever modifications we need to the system at that time to make it conform to our new i18n system. That may mean dropping these `text` attributes from the schema altogether.

**Comment by tie-rack** [comment]
If that's technical debt we're willing to take on: LGTM

**Comment by tie-rack** [comment]
Though.... instead of making the definitely wrong choice (`:instruction/text` and `:voting-method/name`), I'd like to re-affirm my suggestion of taking them out.

**Comment by cap10morgan** [comment]
@tie-rack I hear you, but we need `:instruction/text` before i18n will be ready. I can take out `:voting-method/name` for now, though.

**Comment by tie-rack** [comment]
Okay. Sorry to be such a pest!

**Comment by cap10morgan** [comment]
Not at all. I'm glad you're thinking about these things!

**Pull request merged**